editoast/src/views/authz.rs

```diff
@@ -0,0 +1,229 @@
+use crate::error::Result;
+use crate::models::auth::AuthDriverError;
+use crate::AppState;
+use axum::extract::{Path, State};
+use axum::response::Json;
+use axum::Extension;
+use editoast_authz::roles::BuiltinRoleSet;
+use editoast_authz::BuiltinRole;
+use editoast_derive::EditoastError;
+
+use super::{AuthorizationError, AuthorizerExt};
+
+crate::routes! {
+    "/authz" => {
+        "/list_roles" => list_roles,
+        "/roles/me" => list_current_roles,
+        "/roles/{user_id}" => {
+            list_user_roles,
+            grant_roles,
+            strip_roles,
+        },
+    },
+}
+
+#[derive(serde::Deserialize, utoipa::IntoParams)]
+struct UserIdPathParam {
+    /// A user ID (not to be mistaken for its identity, cf. editoast user model documentation)
+    user_id: i64,
+}
+
+#[derive(Debug, thiserror::Error, EditoastError)]
+#[editoast_error(base_id = "authz")]
+enum AuthzError {
+    #[error("Internal error")]
+    #[editoast_error(status = 500, no_context)]
+    Error(#[from] AuthDriverError),
+    #[error(transparent)]
+    DbError(#[from] editoast_models::DatabaseError), // forwards impl EditoastError
+    #[error("Authorization error")]
+    AuthError(#[from] AuthorizationError),
+}
+
+#[derive(Debug, thiserror::Error, EditoastError)]
+#[editoast_error(base_id = "authz")]
+enum NoSuchUserError {
+    #[error("No user with ID {user_id} found")]
+    #[editoast_error(status = 404)]
+    NoSuchUser { user_id: i64 },
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "authz",
+    responses(
+        (status = 200, description = "List of supported application roles", body = Vec<String>, example = json!(["Superuser", ""])),
+    ),
+)]
+async fn list_roles(State(AppState { role_config, .. }): State<AppState>) -> Json<Vec<String>> {
+    let roles = role_config.application_roles().cloned().collect();
+    Json(roles)
+}
+
+#[derive(serde::Serialize, utoipa::ToSchema)]
+struct Roles {
+    builtin: Vec<String>,
+    application: Vec<String>,
```

Comment:
As discussed, I would rather not expose `application` (yet?) as there is no obvious use case for it.